### PR TITLE
Add `wasm-pack` tests for `net` to CI

### PR DESCRIPTION
**Pull request by littlebenlittle**
Tests to reproduce #484 

Fixed by ~~#485~~ #474

### CODE REVIEW

**Review by ranile — DISMISSED** [review]
Please ensure that the tests aren't failing

File: crates/net/tests/http.rs [hunk 1]
```diff
@@ -3,8 +3,6 @@ use once_cell::sync::Lazy;
 use serde::{Deserialize, Serialize};
 use wasm_bindgen_test::*;
 
-wasm_bindgen_test_configure!(run_in_browser);
```
Comment (ranile) [inline crates/net/tests/http.rs:6]: This is needed for browser tests, please keep this in (and gate it behind `wasm32-unknown-unknown` target)

File: crates/net/tests/query.rs [hunk 2]
```diff
@@ -1,8 +1,6 @@
 use gloo_net::http::QueryParams;
 use wasm_bindgen_test::*;
 
-wasm_bindgen_test_configure!(run_in_browser);
```
Comment (ranile) [inline crates/net/tests/query.rs:4]: same as previous comment

File: Cargo.lock
Comment (ranile) [inline Cargo.lock:1]: This is unrelated, please revert

**Review by ranile — CHANGES_REQUESTED** [review]
You'd want to bring this PR up to date with master to get the latest code for CI

File: crates/net/tests/http.rs (hunk 1 above)
Comment (littlebenlittle) [reply crates/net/tests/http.rs:6]: Clippy doesn't like `#[cfg(target="wasm32-unkonwn-unknown")]` but it seems to work. Let me know if I misunderstood.

Comment (ranile) [reply crates/net/tests/http.rs:6]: I think `target_arch` refers to wasm32 in this case (see https://doc.rust-lang.org/reference/conditional-compilation.html#target_arch). That might be why clippy complains. You want to update CI so clippy is also run on wasm32 target.

You can look at how wasi is cfg-gated in other parts of the code base and do the opposite of it here. If it's `target_os = "wasi"` then you may be able to do `not(target_os = "wasi")` to target browser wasm targets

Comment (littlebenlittle) [reply crates/net/tests/http.rs:6]: So this seems like a quirk of `wasm_bindgen_test`. As far as I can tell there's no way straightforward way to run in both node and browser.

Here's one suggestion: https://github.com/rustwasm/wasm-bindgen/issues/2571#issuecomment-873533119

Added to latest commit.

**Review by ranile — APPROVED** [review]
Having to add a feature is sad but this is the best we can really do. Thanks for the PR!
